pkg/network/vmispec/resource.go

```diff
@@ -0,0 +1,84 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 Red Hat, Inc.
+ *
+ */
+
+package vmispec
+
+import (
+	"fmt"
+
+	k8scorev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	v1 "kubevirt.io/api/core/v1"
+)
+
+type bindingPluginsGetter interface {
+	GetNetworkBindings() map[string]v1.InterfaceBindingPlugin
+}
+
+func CalculateBindingPluginsMemoryOverhead(vmiSpec *v1.VirtualMachineInstanceSpec, pluginGetter bindingPluginsGetter) (resource.Quantity, error) {
+	registeredPlugins := pluginGetter.GetNetworkBindings()
+	if registeredPlugins == nil {
+		return resource.Quantity{}, nil
+	}
+
+	vmiPluginSet, err := getVMIPluginSet(vmiSpec.Domain.Devices.Interfaces, registeredPlugins)
+	if err != nil {
+		return resource.Quantity{}, err
+	}
+
+	return sumPluginsMemoryOverhead(vmiPluginSet), nil
+}
+
+func getVMIPluginSet(ifaces []v1.Interface, registeredPlugins map[string]v1.InterfaceBindingPlugin) (map[string]v1.InterfaceBindingPlugin, error) {
+	vmiPluginSet := make(map[string]v1.InterfaceBindingPlugin)
+
+	for _, iface := range ifaces {
+		if iface.Binding == nil {
+			continue
+		}
+
+		pluginName := iface.Binding.Name
+		if _, exists := vmiPluginSet[pluginName]; exists {
+			continue
+		}
+
+		var ok bool
+		if vmiPluginSet[pluginName], ok = registeredPlugins[pluginName]; !ok {
+			return nil, fmt.Errorf("could not find plugin %q in the registerd plugins", pluginName)
```

Comment:
This check is redundant IMO, even confusing. We already check this at the admitter and can do the same at the controller as a validation.
Here we can just ignore these and remove the error.

Reply:
A plugin is now ignored in case it is not registered.